### PR TITLE
Add ConfigMap support and new deployment

### PR DESCRIPTION
**Pull request by darkowlzz**
Adds support for using configmap for node configuration and new deployment with e2e test.
Also fixes the issue due to which CSI node publish secret was not being deleted.

### CODE REVIEW

File: pkg/storageos/deploy.go [hunk 1]
```diff
@@ -271,6 +279,34 @@ func (s *Deployment) addNodeContainerResources(nodeContainer *corev1.Container)
...
+		TimeoutSeconds:      int32(10),
+		FailureThreshold:    int32(5),
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/v1/health",
+				Port: intstr.IntOrString{Type: intstr.String, StrVal: "api"},
+			},
+		},
+	}
+	nodeContainer.LivenessProbe = &corev1.Probe{
```
[the remark this answers is not in the document]
Comment (domodwyer) [reply pkg/storageos/deploy.go]: Would it be possible to add a readiness probe using a tcp check? In c2, the container is OK to handle requests as soon as the API socket is bound.

File: pkg/storageos/configmap.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	// Always show telemetry and feature options to ensure they're visble.
+	config[disableTelemetryEnvVar] = strconv.FormatBool(spec.DisableTelemetry)
+
+	// TODO: separte CR items for version check and crash reports.  Use
+	// Telemetry to enable/disable everything for now.
+	config[disableVersionCheckEnvVar] = strconv.FormatBool(spec.DisableTelemetry)
+	config[disableCrashReportingEnvVar] = strconv.FormatBool(spec.DisableTelemetry)
+
+	// Features
+	config[disableFencingEnvVar] = strconv.FormatBool(spec.DisableFencing)
```
[the remark this answers is not in the document]
Comment (croomes) [reply pkg/storageos/configmap.go]: yes, we should remove this

File: pkg/storageos/deploy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (domodwyer) [reply pkg/storageos/deploy.go]: I mean we set `nodeContainer.LivenessProbe` twice here!

File: pkg/storageos/configmap.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (darkowlzz) [reply pkg/storageos/configmap.go]: Removed and commented the corresponding blocks in tests.

**Comment by darkowlzz** [comment]
Had to add pod toleration in the etcd operator and cluster pod for proper deployment with tainted nodes.

**Comment by darkowlzz** [comment]
Disabling v2 deployment test on openshift 3.11. We should debug it in the future. Dataplane startup fails due to some reason.
Logs:
```
{"level":"info","msg":"tracing is disabled","time":"2020-01-27T09:36:38.116401981Z"}
{"level":"info","msg":"Jaeger tracer created successfully","time":"2020-01-27T09:36:38.116547205Z"}
{"level":"info","msg":"starting StorageOS","time":"2020-01-27T09:36:38.11663679Z"}
{"env_advertise_address":"10.20.0.16","env_api_bind_address":"","env_api_tls_ca":"","env_bind_address":"","env_bootstrap_namespace":"","env_bootstrap_username":"storageos","env_csi_endpoint":"unix:///var/lib/kubelet/plugins_registry/storageos/csi.sock","env_csi_version":"v1","env_dataplane_daemon_dir":"","env_dataplane_dir":"","env_device_dir":"","env_dial_timeout":"","env_disable_crash_reporting":"false","env_disable_telemetry":"false","env_disable_version_check":"false","env_etcd_endpoints":"etcd-client.default.svc.cluster.local:2379","env_etcd_namespace":"","env_etcd_tls_client_ca":"","env_etcd_tls_client_cert":"","env_etcd_tls_client_key":"","env_etcd_username":"","env_gossip_advertise_address":"","env_gossip_bind_address":"","env_health_grace_period":"","env_health_probe_interval":"","env_health_probe_timeout":"","env_hostname":"localhost","env_internal_api_advertise_address":"","env_internal_api_bind_address":"","env_internal_tls_ca_cert":"","env_internal_tls_node_cert":"","env_internal_tls_node_key":"","env_io_advertise_address":"","env_io_bind_address":"","env_jaeger_endpoint":"","env_jaeger_service_name":"","env_k8s_config_path":"","env_k8s_distribution":"openshift","env_k8s_enable_scheduler_extender":"true","env_k8s_namespace":"storageos","env_log_file":"","env_log_format":"json","env_log_level":"info","env_log_size_limit":"","env_node_capacity_interval":"","env_root_dir":"","env_socket_dir":"","env_supervisor_advertise_address":"","env_supervisor_bind_address":"","level":"info","msg":"environment variables at startup","time":"2020-01-27T09:36:38.116755841Z"}
{"level":"info","msg":"local node StorageOS id: 288b3ea0-de13-4dbb-9e57-f265d8e58036","time":"2020-01-27T09:36:38.117052839Z"}
{"level":"info","msg":"local node Hostname is: localhost","time":"2020-01-27T09:36:38.117125477Z"}
{"error":"no TLS ca certificate provided","level":"warning","msg":"no TLS certificate provided for etcd, communications will not be secured","time":"2020-01-27T09:36:38.117151046Z"}
{"level":"info","msg":"ETCD connection established at: [etcd-client.default.svc.cluster.local:2379]","time":"2020-01-27T09:36:38.130141897Z"}
{"level":"info","msg":"joining cluster: 76021abb-6ab9-4482-ba8a-eba63b3d8e3e","time":"2020-01-27T09:36:38.697960657Z"}
{"cluster_created_at":"2020-01-27T09:35:07Z","cluster_disable_crash_reporting":false,"cluster_disable_telemetry":false,"cluster_disable_version_check":false,"cluster_log_format":"json","cluster_log_level":"info","cluster_updated_at":"2020-01-27T09:35:08Z","cluster_version":2,"level":"info","msg":"fetched cluster configuration","time":"2020-01-27T09:36:38.702223587Z"}
{"cluster_created_at":"2020-01-27T09:35:07Z","cluster_disable_crash_reporting":false,"cluster_disable_telemetry":false,"cluster_disable_version_check":false,"cluster_log_format":"json","cluster_log_level":"info","cluster_updated_at":"2020-01-27T09:35:08Z","cluster_version":2,"level":"info","msg":"switched to cluster log configuration","time":"2020-01-27T09:36:38.702519111Z"}
{"level":"info","msg":"this cluster is configured to send anonymous usage data to help us develop StorageOS (https://docs.storageos.com/docs/reference/telemetry)","time":"2020-01-27T09:36:38.702662071Z"}
{"level":"warning","msg":"no CA certificate provided, falling back to using automatically generated TLS data","time":"2020-01-27T09:36:38.705131608Z"}
{"level":"error","msg":"stopped persisting node capacity data","time":"2020-01-27T09:36:38.705632483Z"}
{"error":"stat /var/lib/storageos/volumes: transport endpoint is not connected","level":"error","msg":"dataplane startup failed","time":"2020-01-27T09:36:38.70704915Z"}
{"level":"info","msg":"shutting down","time":"2020-01-27T09:36:38.707742998Z"}
```

**Pull request merged**